frame/assets/src/lib.rs

```diff
@@ -569,45 +589,207 @@ pub mod pallet {
 			Self::do_force_create(id, owner, is_sufficient, min_balance)
 		}
 
-		/// Destroy a class of fungible assets.
+		/// Start the process of destroying a class of fungible asset
+		/// start_destroy is the first in a series of extrinsics that should be called, to allow
+		/// destroying an asset.
 		///
 		/// The origin must conform to `ForceOrigin` or must be Signed and the sender must be the
 		/// owner of the asset `id`.
 		///
 		/// - `id`: The identifier of the asset to be destroyed. This must identify an existing
-		/// asset.
-		///
-		/// Emits `Destroyed` event when successful.
-		///
-		/// NOTE: It can be helpful to first freeze an asset before destroying it so that you
-		/// can provide accurate witness information and prevent users from manipulating state
-		/// in a way that can make it harder to destroy.
-		///
-		/// Weight: `O(c + p + a)` where:
-		/// - `c = (witness.accounts - witness.sufficients)`
-		/// - `s = witness.sufficients`
-		/// - `a = witness.approvals`
-		#[pallet::weight(T::WeightInfo::destroy(
-			witness.accounts.saturating_sub(witness.sufficients),
- 			witness.sufficients,
- 			witness.approvals,
- 		))]
-		pub fn destroy(
+		///   asset.
+		///
+		/// Assets must be freezed before calling start_destroy.
+		#[pallet::weight(T::WeightInfo::start_destroy())]
+		pub fn start_destroy(
+			origin: OriginFor<T>,
+			#[pallet::compact] id: T::AssetId,
+		) -> DispatchResult {
+			let maybe_check_owner = match T::ForceOrigin::try_origin(origin) {
+				Ok(_) => None,
+				Err(origin) => Some(ensure_signed(origin)?),
+			};
+			let _ = Asset::<T, I>::try_mutate_exists(
+				id,
+				|maybe_details| -> Result<(), DispatchError> {
+					let mut details = maybe_details.as_mut().ok_or(Error::<T, I>::Unknown)?;
+					if let Some(check_owner) = maybe_check_owner {
+						ensure!(details.owner == check_owner, Error::<T, I>::NoPermission);
+					}
+					ensure!(details.is_frozen, Error::<T, I>::BadWitness);
+					details.status = AssetStatus::Destroying;
+					// TODO: Remove previlleged roles. How?
```

Comment:
For starters, you can add into every non-destroying function:

```rust
ensure!(details.status == AssetStatus::Live, Error::<T, I>::AssetNotLive);
```

This will prevent even privileged accounts from doing anything as every call will fail. A better solution, and a separate PR, would be to make the `owner`, `admin`, `issuer`, and `freezer` `Option<AccountId>` and then set them to `None`.

Reply:
For now, I've added the AssetNotLive check to the other extrinsics.